sky/clouds/vast.py

```diff
@@ -0,0 +1,296 @@
+""" Vast Cloud. """
+
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple, Union
+
+from sky import clouds
+from sky.clouds import service_catalog
+from sky.utils import resources_utils
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib
+
+_CREDENTIAL_FILES = [
+    'config.toml',
+]
+
+
+@clouds.CLOUD_REGISTRY.register
+class Vast(clouds.Cloud):
+    """ Vast GPU Cloud
+
+    _REPR | The string representation for the Vast GPU cloud object.
+    """
+    _REPR = 'Vast'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE:
+            ('Spot is not supported, as vast API does not implement spot.'),
+        clouds.CloudImplementationFeatures.MULTI_NODE:
+            ('Multi-node not supported yet, as the interconnection among nodes '
+             'are non-trivial on Vast.'),
+        ## TODO: These are different from the fluffy cloud
+        clouds.CloudImplementationFeatures.CUSTOM_DISK_TIER:
+            ('Customizing disk tier is not supported yet on Vast.'),
+        clouds.CloudImplementationFeatures.STORAGE_MOUNTING:
+            ('Mounting object stores is not supported on Vast. To read data '
+             'from object stores on Vast, use `mode: COPY` to copy the data '
+             'to local disk.'),
```

Comment:
same, is this not supported as well? This seems only not enabled in runpod for now, which I suppose is due to its pod nature for provisioned "VM"s. Is vast ai also provide pods to the user?

Reply:
see 98e3db50e9110fa0480398dc0e6e647c0f0fd521

Reply:
Just want to make sure, it sounds like you are referring to cloud storage from VastAI? IIUC this feature actually means can you mount storages from any cloud (e.g. S3, GCS) in the VastAI clusters, and it should be supported as long as some mounting utils is supported. Though if VastAI using the same container fashion provisioning, it might not work as well. cc @romilbhardwaj - what is the reason runpod not support this feature?

Reply:
users are succeeding in this through our container model but we don't have any "official" support. We just announced/released VM support today and with it users will be able to do more conventional mounting of remote stores. If this is really a deal breaker I can work on making it officially supported everywhere somehow but that's a larger project.

Reply:
I see. If that is the case then LGTM!